custom/samveg/case_importer/validators.py

```diff
@@ -103,9 +103,10 @@ def run(cls, row_num, raw_row, fields_to_update, **kwargs):
         return fields_to_update, error_messages
 
 
-class UploadLimitValidator(BaseOperation):
+class UploadLimitValidator(BaseRowOperation):
     @classmethod
-    def run(cls, row_num, raw_row, fields_to_update, import_context):
+    def run(cls, row_num, raw_row, fields_to_update, **kwargs):
+        import_context = kwargs['import_context']
```

Comment:
I wouldn't call this an optional argument (as documented in `:param kwargs: Optional additional kwargs passed for a specific operation`). It will raise `KeyError` if it is not provided. If you want to make it required for this validator you could do something like
```py
def run(cls, row_num, raw_row, fields_to_update, *, import_context, **kwargs):
```
However, even that has a smell because the thing that calls `run(...)` always provides `import_context` and never provides any other arguments, so why not just make the signature for all operations require it?
```suggestion
    def run(cls, row_num, raw_row, fields_to_update, import_context):
```
It seems like you are planning for future (and as yet unknown) requirements by accepting `**kwargs`. This is an anti-pattern. Better to not make the code more flexible than it needs to be because that surprisingly often makes it harder to change in the future.

Reply:
> It will raise KeyError if it is not provided

That is what I intended. 

I am almost done with changes I need for this feature and import_context is the only kwarg I need so makes sense to make that mandatory.
https://github.com/dimagi/commcare-hq/pull/30921/commits/f9da7eaa123af1ee707675fa2265b922c61be319

> It seems like you are planning for future (and as yet unknown) requirements by accepting **kwargs. This is an anti-pattern. Better to not make the code more flexible than it needs to be because that surprisingly often makes it harder to change in the future

Thanks for this note. A good reminder for me since I tend to do that.